core/storage/base_model/gae_models_test.py

```diff
@@ -349,15 +349,25 @@ def test_get_multi_versions_errors(self):
             'At least one version number is invalid'):
             TestVersionedModel.get_multi_versions('model_id1', [1, 1.5, 2])
 
-    # Note: This test for the generic base model is written at last since some
-    # unrelated tests fail if this test is written at the beginning.
-    def test_raise_exception_by_mocking_collision(self):
-        with self.assertRaisesRegexp(
-            Exception, 'New id generator is producing too many collisions.'):
-            # Swap dependent method get_by_id to simulate collision every time.
-            with self.swap(
-                base_models.BaseModel, 'get_by_id',
-                types.MethodType(
-                    lambda x, y: True,
-                    base_models.BaseModel)):
-                base_models.BaseModel.get_new_id('exploration')
+
+class TestBaseModel(base_models.BaseModel):
+    """Model that inherits BaseModel for testing. This is required as BaseModel
+    gets subclassed a lot in other tests and that can create unexpected errors.
+    """
+    pass
+
+
+class BaseModelTests(test_utils.GenericTestBase):
+
+    def test_create_raises_error_when_many_id_collisions_occur(self):
+
+        # Swap dependent method get_by_id to simulate collision every time.
+        get_by_id_swap = self.swap(
+            TestBaseModel, 'get_by_id', types.MethodType(
+                lambda _, __: True, TestBaseModel))
```

Comment:
Here and elsewhere: should the second underscore (after the comma) be just one underscore, not two?

Reply:
Lambda cannot accept arguments with the same name I think. I even tried it, it gives an error.

Reply:
Oh I see, makes sense. OK, sounds good, thanks!